### PR TITLE
 refactor(cli): change to cobra cli

### PR DESCRIPTION
**Pull request by sergeylanzman**
- [X] refactor command package
- [X] separate colorize from gateway client
- [X] convert application command to cobra 
- [X] convert pipeline command to cobra
- [X] delete old command
- [x] fix application tests
- [x] fix pipeline tests
- [x] split run command to functions

### CODE REVIEW

**Comment by sergeylanzman** [comment]
@jtk54 Done, it's work with cobra now.
Need more improvement examples and description(long, short).
I think need change tests to table testing(less copy paste code in tests)
We can add now bash autocomplete and autogenerate docs and other cobra options.

File: cmd/gateclient/client.go [hunk 1]
```diff
@@ -161,14 +139,14 @@ func (m *ApiMeta) Process(args []string) ([]string, error) {
...
-		return args, err
+		util.UI.Error(fmt.Sprintf("OAuth2 Authentication failed."))
+		return nil, err
 	}
 
 	client, err := m.InitializeClient()
 	if err != nil {
-		m.Ui.Error(fmt.Sprintf("Could not initialize http client, failing."))
-		return args, err
+		util.UI.Error(fmt.Sprintf("Could not initialize http client, failing."))
```
Comment (jtk54) [inline cmd/gateclient/client.go]: May be worth it at some point to create a `UI.Errorf()` function at some point to clean up the `fmt.Sprintf()s`. Not necessary in this PR though.

File: cmd/gateclient/client.go [hunk 2]
```diff
@@ -249,7 +217,7 @@ func (m *ApiMeta) InitializeClient() (*http.Client, error) {
 		}
 	} else if auth != nil && auth.Enabled && auth.Basic != nil {
 		if !auth.Basic.IsValid() {
-			return nil, errors.New("Incorrect Basic auth configuration. Must include username and password.")
+			return nil, errors.New("incorrect Basic auth configuration. Must include username and password")
```
Comment (jtk54) [inline cmd/gateclient/client.go]: Is there a particular reason for the punctuation change?

File: cmd/gateclient/client.go [hunk 3]
```diff
@@ -280,7 +248,7 @@ func (m *ApiMeta) Authenticate() error {
 		OAuth2 := auth.OAuth2
 		if !OAuth2.IsValid() {
 			// TODO(jacobkiefer): Improve this error message.
-			return errors.New("Incorrect OAuth2 auth configuration.")
+			return errors.New("incorrect OAuth2 auth configuration")
```
Comment (jtk54) [inline cmd/gateclient/client.go]: Same here.

File: cmd/pipeline/delete.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+var (
+	deletePipelineShort   = "Delete the provided pipeline"
+	deletePipelineLong    = "Delete the provided pipeline"
+	deletePipelineExample = `
+	usage: spin pipeline delete [options]
+
+	Delete the provided pipeline
+
+    --application: Spinnaker application the pipeline lives in
```
Comment (jtk54) [inline cmd/pipeline/delete.go]: Is there a way to generate the help text from the `PersistentFlags()`?

File: cmd/root.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
+package cmd
+
+import (
+	"io"
+
+	"github.com/spinnaker/spin/util"
+
+	"github.com/spinnaker/spin/cmd/pipeline"
+
+	"github.com/spinnaker/spin/cmd/application"
```
Comment (jtk54) [inline cmd/root.go]: Why the empty lines here?

File: cmd/gateclient/client.go (hunk 2 above)
Comment (sergeylanzman) [reply cmd/gateclient/client.go]: not critical
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

File: cmd/pipeline/delete.go (hunk 4 above)
Comment (sergeylanzman) [reply cmd/pipeline/delete.go]:
```
# spin help
Global Options:

		--gate-endpoint               Gate (API server) endpoint.
		--no-color                    Removes color from CLI output.
		--insecure=false              Ignore certificate errors during connection to endpoints.
		--quiet=false                 Squelch non-essential output.
		--output <output format>      Formats CLI output.

Usage:
   [command]

Available Commands:
  application
  help        Help about any command
  pipeline

Flags:
      --config string          config file (default is $HOME/.spin/config)
      --gate-endpoint string   Gate (API server) endpoint. Default http://localhost:8084
  -h, --help                   help for this command
      --insecure               Ignore Certificate Errors
      --no-color               Disable color (default true)
      --output string          Configure output formatting
      --quiet                  Squelch non-essentual output

Use " [command] --help" for more information about a command.
```

**Comment by jtk54** [comment]
Agreed on the test code duplication, I had a PR open and then shelved it to focus on other things. I'm excited about the cobra improvements. I'll take another look today and test it out.

File: cmd/pipeline/get.go [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+var (
+	getPipelineShort   = "Get the pipeline with the provided name from the provided pipeline"
+	getPipelineLong    = "Get the specified pipeline"
+	getPipelineExample = `
+	usage: spin pipeline get [options]
+
+List the pipelines for the provided pipeline
+
+--pipeline: Pipeline the pipeline lives in
```
Comment (jtk54) [inline cmd/pipeline/get.go]: This should be `--application: Application the pipeline ...`.

File: cmd/pipeline/get.go [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+	name        string
+}
+
+var (
+	getPipelineShort   = "Get the pipeline with the provided name from the provided pipeline"
+	getPipelineLong    = "Get the specified pipeline"
+	getPipelineExample = `
+	usage: spin pipeline get [options]
+
+List the pipelines for the provided pipeline
```
Comment (jtk54) [inline cmd/pipeline/get.go]: `provided application`.

File: cmd/pipeline/get.go [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+type GetOptions struct {
+	*pipelineOptions
+	output      string
+	application string
+	name        string
+}
+
+var (
+	getPipelineShort   = "Get the pipeline with the provided name from the provided pipeline"
```
Comment (jtk54) [inline cmd/pipeline/get.go]: s/provided pipeline/provided application/

**Pull request merged**